packages/ember-glimmer/lib/utils/iterable.js

```diff
@@ -175,10 +175,18 @@ class EachInIterable {
     }
 
     let typeofIterable = typeof iterable;
-
     if (iterable && (typeofIterable === 'object' || typeofIterable === 'function')) {
-      let keys = Object.keys(iterable);
-      let values = keys.map(key => iterable[key]);
+      let keys = [];
+      let values = [];
+      if (HAS_NATIVE_MAP && iterable instanceof Map) {
```

Comment:
Also, its likely possible to create a custom iterator object that avoids eagerly iterating all the items (like I did in that snippet), but we would need to pester some folks to get some clarity on the "right" way to do it.